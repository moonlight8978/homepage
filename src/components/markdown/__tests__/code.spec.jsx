import React from 'react'
import { render } from '@testing-library/react'

import Code from '../code'

test('renders code block corresponding to language', () => {
  const language = 'javascript'
  const code = `import React from "react"`
  const { container } = render(<Code language={language} value={code} />)
  expect(container.firstChild).toMatchSnapshot()
})

test('renders plaintext if language is not supported', () => {
  const language = 'invalid'
  const code = `This is an invalid language`
  const { container } = render(<Code language={language} value={code} />)
  expect(container.firstChild).toMatchSnapshot()
})
