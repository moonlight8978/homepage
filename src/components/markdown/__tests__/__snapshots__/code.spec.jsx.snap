// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders code block corresponding to language 1`] = `
<pre
  class="container"
>
  <code
    class="language-javascript code"
  >
    import React from "react"
  </code>
</pre>
`;

exports[`renders plaintext if language is not supported 1`] = `
<pre
  class="container"
>
  <code
    class="language-plaintext code"
  >
    This is an invalid language
  </code>
</pre>
`;
